### ISSUE TITLE
False positive: parameter must be non-null in inner class constructor 

### ISSUE THREAD

**Issue by ahubold** [spotbugs/sonar-findbugs#213]
## Issue Description

Sonar Findbugs shows an issue for rule findbugs:NP_PARAMETER_MUST_BE_NONNULL_BUT_MARKED_AS_NULLABLE that an inner class constructor parameter must be non-null, even if the parameter is already annotated as NonNull. See code example below.

## Environment

| Component          | Version |
| ------------------ | ------- |
| SonarQube          | 6.7.3  |
| Sonar-FindBugs     | 3.8.0   |
| Maven              | 3.5  |
| Java               | 1.8|

## Code

```java
package com.example;

import edu.umd.cs.findbugs.annotations.NonNull;
import edu.umd.cs.findbugs.annotations.Nullable;

import java.util.Objects;

public class Outer {

  private class Inner {
    private final String a;
    private final Object b;
    private final Object c;

    Inner(@NonNull String a,
          @Nullable Object b,
          @Nullable Object c) {
      this.a = Objects.requireNonNull(a); // <--- a must be non-null but is marked as nullable
      this.b = b;
      this.c = c;
    }

  }
}
```

**Comment by KengoTODA** [comment]
This bug comes from spotbugs side, and is same with hhttps://github.com/spotbugs/spotbugs/issues/567

So I recommend you to issue not in sonar-findbugs project but spotbugs project.

**Comment by ahubold** [comment]
Thank you. Are you saying that this bug will be fixed with https://github.com/spotbugs/spotbugs/issues/567 or should I open a new issue for spotbugs?

**Comment by ahubold** [comment]
I've opened a separate Spotbugs issue now: https://github.com/spotbugs/spotbugs/issues/772

**Comment by KengoTODA** [comment]
Thank you! Then I will close this issue in sonar-findbugs. :+1:
